ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestSuiteRunner.scala

```diff
@@ -129,9 +126,15 @@ final class LedgerTestSuiteRunner(
       .flatMap(suite => suite.tests.map(suite -> _))
       .zipWithIndex
     Source(tests)
+      .throttle(3, 1.second)
       .mapAsyncUnordered(concurrentTestRuns) {
-        case ((suite, test), index) =>
-          run(test, suite.session).map(testResult => (suite, test, testResult) -> index)
+        case ((suite, test), index) => {
+          if (shouldRun(test.name)) {
+            run(test, ledgerSession).map(testResult => (suite, test, testResult) -> index)
+          } else {
+            Future.successful((suite, test, Right(Result.Skipped("Skipped"))) -> index)
+          }
+        }
```

Comment:
```suggestion
        case ((suite, test), index) if shouldRun(test.name) =>
          run(test, ledgerSession).map(testResult => (suite, test, testResult) -> index)
        case ((suite, test), index) =>
          Future.successful((suite, test, Right(Result.Skipped("Skipped"))) -> index)
        }
```
You can avoid a bit of nesting using guards on the case expression instead of a nested `if`. Furthermore, `case`s don't need braces (not even multi-line ones). Feel free to discard this suggestion, I trust your judgement.